From 785a490556b549493a6a25409f16026ce0959b02 Mon Sep 17 00:00:00 2001
From: Longlong Xia <xialonglong@kylinos.cn>
Date: Sun, 9 Aug 2026 22:29:28 +0800
Subject: [PATCH 12408/15333] Input: elan_i2c - use device-id/acpi.h for ACPI
 IDs

elan-i2c-ids.h only needs struct acpi_device_id from the ACPI
device ID definitions. The MODULE_DEVICE_TABLE() user already includes
<linux/module.h>.

Include <linux/device-id/acpi.h> instead of the broader
<linux/mod_devicetable.h> header.

Assisted-by: Codex:GPT-5
Signed-off-by: Longlong Xia <xialonglong@kylinos.cn>
Link: https://patch.msgid.link/20260809142928.4031270-1-xialonglong2025@163.com
Signed-off-by: Dmitry Torokhov <dmitry.torokhov@gmail.com>
---
 include/linux/input/elan-i2c-ids.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/input/elan-i2c-ids.h b/include/linux/input/elan-i2c-ids.h
index 51cca17ee94c..874bf0ab500c 100644
--- a/include/linux/input/elan-i2c-ids.h
+++ b/include/linux/input/elan-i2c-ids.h
@@ -18,7 +18,7 @@
 #ifndef __ELAN_I2C_IDS_H
 #define __ELAN_I2C_IDS_H
 
-#include <linux/mod_devicetable.h>
+#include <linux/device-id/acpi.h>
 
 static const struct acpi_device_id elan_acpi_id[] = {
 	{ "ELAN0000", 0 },
-- 
2.55.0

